### PR TITLE
grv: 0.1.2 -> 0.2.0

### PR DESCRIPTION
**Pull request by dtzWill**
Needs newer libgit2 than we have,
so added variant for 0.27.

Our libgit2 has comments saying not to
update without other packages along with,
but when I looked earlier those didn't
support newer libgit2 :(.
That may have changed.

Anyway, yay grv update!

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dtzWill** [comment]
Ah, it's libgit2-glib holding us back I think: https://gitlab.gnome.org/GNOME/libgit2-glib/issues/1

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -9339,6 +9339,10 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) Security;
   };
 
+  libgit2_0_27 = callPackage ../development/libraries/git2/0.27.nix {
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: Does `blsd` also support `libgit2_0_27`, then `libgit2_0_25` can be dropped.

Comment (Mic92) [reply pkgs/top-level/all-packages.nix]: Or maybe libgit2_0_26 which is what our libgit2 defaults to.

**Comment by Mic92** [comment]
glib bindings can be bumped like this: https://gitlab.gnome.org/smcv/libgit2-glib/commit/f6d7a149b40a7b131852f4daf8b428bbc993b12e

**Comment by Mic92** [comment]
If glib is the only blocker and can be fixed with the patch, then we can have a single version of libgit2 in nixpkgs.

**Comment by dtzWill** [comment]
Sounds good to me, is there someone who can approve/check? :)

**Comment by andir** [comment]
It looks fine and grv works as expected.

The bumped deps don't seem to cause an outbreak of breakage.

@dtzWill one minor nitpick is that `grv` should be reffered to as `gitAndTools.grv`.


I've built the affected packages locally. They look ok. Lets see what ofBorg says before I merge this just to be sure (about aarch64 and darwin…)
@GrahamcOfBorg build julia_05 gitAndTools.grv blsd (I expect julia to time out...)

**Comment by andir** [comment]
julia fails it's libgit test with the new version:

```
Exception running test libgit2 :
On worker 8:
LoadError: There was an error during testing
 in record at ./test.jl:397
 in do_test at ./test.jl:281
 in #351 at /build/source/test/libgit2.jl:209
 in mktempdir at ./file.jl:286
 in mktempdir at ./file.jl:284
 in macro expansion at ./util.jl:230 [inlined]
 in runtests at /build/source/test/testdefs.jl:7
 in #16 at /build/source/test/runtests.jl:44
 in #645 at ./multi.jl:1421
 in run_work_thunk at ./multi.jl:1001
 in macro expansion at ./multi.jl:1421 [inlined]
 in #644 at ./event.jl:68
while loading /build/source/test/libgit2.jl, in expression starting on line 130
ERROR: LoadError: Some tests exited with errors.
 in (::##11#19)() at /build/source/test/runtests.jl:72
 in cd(::##11#19, ::String) at ./file.jl:59
while loading /build/source/test/runtests.jl, in expression starting on line 26
make[1]: *** [Makefile:12: all] Error 1
make[1]: Leaving directory '/build/source/test'
make: *** [Makefile:577: testall] Error 2
```

**Comment by Mic92** [comment]
Actually it does not look like this version of julia receives updates from upstream. Should we just drop julia_4 and julia_5 instead of putting effort into this?

**Comment by Mic92** [comment]
I think this pull request can be merged now.

**Pull request merged**